frontend/packages/shipwright-plugin/src/components/build-tabsection/BuildOverview.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import { Flex, FlexItem } from '@patternfly/react-core';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import {
+  ResourceLink,
+  resourcePath,
+  SidebarSectionHeading,
+} from '@console/internal/components/utils';
+import { referenceForModel } from '@console/internal/module/k8s/k8s';
+import { OverviewItem } from '@console/shared';
+import { BUILDRUN_TO_RESOURCE_MAP_LABEL } from '../../const';
+import { BuildModel } from '../../models';
+import { Build, BuildRun } from '../../types';
+import BuildRunItem from './BuildRunItem';
+import StartBuildButton from './StartBuildButton';
+import TriggerLastBuildButton from './TriggerLastBuildButton';
+
+const MAX_VISIBLE = 3;
+
+type BuildsOverviewProps = {
+  item: OverviewItem & {
+    builds?: Build[];
+    buildRuns?: BuildRun[];
+  };
+};
+
+const BuildsOverview: React.FC<BuildsOverviewProps> = ({ item: { builds, buildRuns, obj } }) => {
+  const { t } = useTranslation();
+  const buildRunsforResource = _.filter(buildRuns, (buildRun) => {
+    return (
+      buildRun.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL] ===
+      obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL]
+    );
+  });
```

Comment:
Labels are optional, so we should ensure that we don't compare undefined === undefined and then show all BuildRuns that doesn't have a label.

```suggestion
  const resourceLabel = obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL]
  const buildRunsforResource = resourceLabel ? buildRuns.filter((buildRun) => {
    return (
      resourceLabel === buildRun.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL]
    );
  }) : [];
```